### DISCUSSION TITLE
data-sources.json is not conform to JSON specification, some special characters are not escaped

### DISCUSSION THREAD

**Discussion by aisbergde** [dbeaver/dbeaver#12501] [DBeaver Development]
I try to generate the virtual key definitions in `data-sources.json` by exporting JSON from SQL Server (where I have all these definitions). But The SQL Server can only create JSON which is conform to the JSON specification, and this JSON will escape some characters.

For general info about escaping and encoding in JSON, see Section 2.5 of the JSON RFC - https://www.ietf.org/rfc/rfc4627.txt.

"/" are not escaped in `data-sources.json` and it is not possible to create this kind of not conform JSON by exporting from SQL Server:

https://docs.microsoft.com/en-us/sql/relational-databases/json/how-for-json-escapes-special-characters-and-control-characters-sql-server?view=sql-server-ver15

dbeaver:

```json
	"virtual-models": {
		"microsoft-177b4c65f69-4b8d1d14f182fa5d": {
			"dhw_self": {
				"repo": {
					":ExecutionLog_gross": {
						"constraints": {
							"PK_ExecutionLog_gross": {
								"type": "vpk",
								"attributes": [
									"id"
								]
							}
						},
						"foreign-keys": [
							{
								"entity": "database://microsoft-177b4c65f69-4b8d1d14f182fa5d/dhw_self/org.jkiss.dbeaver.ext.mssql.model.SQLServerSchema/repo/org.jkiss.dbeaver.ext.mssql.model.SQLServerTable/ExecutionLog",
								"constraint": "PK_ExecutionLog",
								"attributes": {
									"id": "id"
								}
							}
						]

```

conform to specification:

```json
"entity": "database:\/\/microsoft-177b4c65f69-4b8d1d14f182fa5d\/dhw_self\/org.jkiss.dbeaver.ext.mssql.model.SQLServerSchema\/repo\/org.jkiss.dbeaver.ext.mssql.model.SQLServerTable\/Index_Settings",
```

#### System information: 
- Operating system (distribution) and version Windows 10
- DBeaver Version 21.0.0.202102281712
- Additional extensions

#### Connection specification:
- Database name and version SQL Server 2019
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:

I did not find how to _**not escape**_ special characters to be able to create the dbaever json.

But it looks like dbeaver can read and use the valid json which I can create. If this is the case than it is a workaround for me to create not identical json, but just to create json that works.

Better would be: #11626 (if dbeaver could export and import virtual keys to and from database), Then nobody needs to try to create content of `data-sources.json`

**Comment by serge-rider** [comment]
We use Gson in lenient mode to save/load JSON configuration. It is pretty standard.
As far as I know it doesn't escape slashes even in strict mode.

It is said that slash escaping is optional: https://stackoverflow.com/questions/29396608/how-to-escape-slashes-in-gson

**Comment by aisbergde** [comment]
Yes, I think it is not an issue. As I wrote

>But it looks like dbeaver can read and use the valid json which I can create. If this is the case than it is a workaround for me to create not identical json, but just to create json that works.
